### PR TITLE
Only run tests for affected languages in PRs

### PR DESCRIPTION
**Pull request by MindFreeze**
This should speed up PR checks and reduce the load on CI

### CODE REVIEW

**Review by tetele — COMMENTED** [review]
I like the approach, but I'd add a bit more logic: if **ANY** file was modified in a PR outside of `sentences`, `responses` and `tests`, then run a full test. Otherwise, just run the tests for the modified languages.

I think it's safer that way

**Comment by MindFreeze** [comment]
Alright. I made it run full tests if a language can't be matched or files outside of translations a changed. Tweaked a BG file to test if it matched it properly and it did.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -32,6 +33,46 @@ jobs:
...
+
+          if [ -n "$LANGUAGES" ]; then
+            echo "languages=$LANGUAGES" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Run tests (PR)
+        if: github.event_name == 'pull_request'
+        run: |
+          if [ -n "${{ steps.extract-languages.outputs.languages }}" ]; then
+            script/test --language ${{ steps.extract-languages.outputs.languages }}
```
Comment (balloob) [inline .github/workflows/ci.yaml:68]: Does this arg work with multiple languages ?

Comment (tetele) [reply .github/workflows/ci.yaml:68]: It seems to work with comma separated values: `pytest tests --language he,ru`

Comment (tetele) [reply .github/workflows/ci.yaml:68]: Yes, code confirms that's the way to go about it https://github.com/home-assistant/intents/blob/f85c1f3c74fed771a0cdabded8f2cc8e0bface47/tests/conftest.py#L27

Comment (MindFreeze) [reply .github/workflows/ci.yaml:68]: one of my first CI runs wrongly matched all languages and this command worked

**Review by balloob — APPROVED** [review]

**Pull request merged**